packages/playground/cli/src/xdebug-path-mappings.ts

```diff
@@ -0,0 +1,419 @@
+			if (!fs.existsSync(dirname)) {
+				if (ides.length > 1) return;
+
+				fs.mkdirSync(dirname);
+			}
+			fs.writeFileSync(
+				configFilePath,
+				'<?xml version="1.0" encoding="UTF-8"?>\n<project version="4">\n</project>'
+			);
+		}
+
+		const contents = fs.readFileSync(configFilePath, 'utf8');
+		const xmlParser = new XMLParser(xmlParserOptions);
+		// NOTE: Using an IIFE so `config` can remain const.
+		const config = (() => {
+			try {
+				return xmlParser.parse(contents, true);
+			} catch (e) {
+				logger.error(e);
+				throw new Error(
+					'There was an error parsing PhpStorm workspace.xml.'
+				);
+			}
+		})();
+
+		let projectElement = config?.find((c: any) => c?.project !== undefined);
+		if (projectElement) {
+			const projectVersion = projectElement[':@']?.version;
+			if (projectVersion === undefined) {
+				throw new Error(
+					'PhpStorm IDE integration only supports <project version="4"> in workspace.xml, ' +
+						'but the <project> configuration has no version number.'
+				);
+			} else if (projectVersion !== '4') {
+				throw new Error(
+					'PhpStorm IDE integration only supports <project version="4"> in workspace.xml, ' +
+						`but we found a <project> configuration with version "${projectVersion}".`
+				);
+			}
+		}
+		if (projectElement === undefined) {
+			projectElement = {
+				project: [],
+				':@': { version: '4' },
+			};
+			config.push(projectElement);
+		}
+
+		let componentElement = projectElement.project.find(
+			(c: any) =>
+				c?.component !== undefined && c?.[':@']?.name === 'PhpServers'
+		);
+		if (componentElement === undefined) {
+			componentElement = {
+				component: [],
+				':@': { name: 'PhpServers' },
+			};
+			projectElement.project.push(componentElement);
+		}
+
+		let serversElement = componentElement.component.find(
+			(c: any) => c?.servers !== undefined
+		);
+		if (serversElement === undefined) {
+			serversElement = { servers: [] };
+			componentElement.component.push(serversElement);
+		}
+
+		const serverElementIndex = serversElement.servers.findIndex(
+			(c: any) => c?.server !== undefined && c?.[':@']?.name === name
+		);
+		if (serverElementIndex === -1) {
+			serversElement.servers.push(serverElement);
+		} else {
+			serversElement.servers[serverElementIndex] = serverElement;
+		}
+
+		const xmlBuilder = new XMLBuilder(xmlBuilderOptions);
+		const xml = xmlBuilder.build(config);
+
+		fs.writeFileSync(configFilePath, xml);
+	}
+
+	// VSCode
+	if (ides.includes('vscode')) {
+		const configuration = {
+			name: name,
+			type: 'php',
+			request: 'launch',
+			port: 9003,
+			pathMappings: mappings.reduce((acc, mount) => {
+				acc[
+					mount.vfsPath
+				] = `\${workspaceFolder}/${mount.hostPath.replace(
+					/^\.\/?/,
+					''
+				)}`;
+				return acc;
+			}, {} as Record<string, string>),
+		};
+
+		const configFilePath = path.join(process.cwd(), '.vscode/launch.json');
+
+		if (!fs.existsSync(configFilePath)) {
```

Comment:
When would that be the case? Should we just bale out if the config isn't there? It could indicate the user forgot to cd to the right directory. We could offer a way to force it and create a default config in this case.

Reply:
> When would that be the case?

New vscode projects don't automatically have a `launch.json` file. I don't know about PhpStorm projects, but I think that `workspace.xml` is not automatically created.

What if we see an `.idea` folder or a `.vscode` folder? Should that be enough to create the default configs?

Reply:
> What if we see an .idea folder or a .vscode folder? Should that be enough to create the default configs?

I think so

Maybe instead of another flag we could ask yes/no and have one collective flag for "-yes"